lib/workers/repository/init/merge.ts

```diff
@@ -17,26 +17,39 @@ import {
 import { logger } from '../../../logger';
 import * as npmApi from '../../../modules/datasource/npm';
 import { platform } from '../../../modules/platform';
+import * as memCache from '../../../util/cache/memory';
 import { getCache } from '../../../util/cache/repository';
 import { readLocalFile } from '../../../util/fs';
 import { getFileList } from '../../../util/git';
 import * as hostRules from '../../../util/host-rules';
 import type { RepoFileConfig } from './types';
 
 export async function detectRepoFileConfig(): Promise<RepoFileConfig> {
+  const cacheKey = 'detectRepoFileConfig';
+  let repoFileConfig = memCache.get<RepoFileConfig>(cacheKey);
+  if (repoFileConfig) {
+    return repoFileConfig;
+  }
   const cache = getCache();
   let { configFileName } = cache;
   if (configFileName) {
-    let configFileParsed = (await platform.getJsonFile(configFileName))!;
+    let configFileRaw: string | undefined = (await platform.getRawFile(
```

Comment:
why now using raw instead of json ?

Reply:
Also the web api requests are also already cached in memory cache, so no extra api calls are done  if this function is called twice

Reply:
all getJson does is parse the return value of getRaw. and we need the raw data for the migration flow.

Reply:
this saves us using the repo cache, which wont help us much due to https://github.com/renovatebot/renovate/pull/16664#issuecomment-1207290744

Reply:
if the `memCache` is not needed for performance it can be removed if we wish so. 
the important bit here is the raw data which will be used later on.